### PR TITLE
Add documentation for [JS-SDK] ParseQuery.withCount

### PR DESCRIPTION
**Pull request by BufferUnderflower**
See:
https://github.com/parse-community/Parse-SDK-JS/issues/865
https://github.com/parse-community/Parse-SDK-JS/pull/868

### CODE REVIEW

**Comment by dplewis** [comment]
Can you add a disclaimer? Count operations can be slow and expensive.

**Comment by dplewis** [comment]
Looks good, I’ll add this to the PHP SDK as well.

File: _includes/js/queries.md [hunk 1]
```diff
@@ -57,6 +57,25 @@ You can skip the first results by setting `skip`. In the old Parse hosted backen
 query.skip(10); // skip the first 10 results
 ```
 
+If you want to know the total number of rows in table satisfying your query, for  e.g. pagination purposes - you can use `withCount` (`false` by default). **Note** Enabling this flag will change the structure of response, see example below.
```
Comment (TomWFox) [inline _includes/js/queries.md]:
```suggestion
If you want to know the total number of rows in a table satisfying your query, for  e.g. pagination purposes - you can use `withCount` (`false` by default). 

**Note:** Enabling this flag will change the structure of response, see the example below.
```

File: _includes/js/queries.md [hunk 2]
```diff
@@ -57,6 +57,25 @@ You can skip the first results by setting `skip`. In the old Parse hosted backen
...
+
+Let's say you have 200 rows in `GameScore` table:
```
Comment (TomWFox) [inline _includes/js/queries.md]:
```suggestion
Let's say you have 200 rows in a table called `GameScore`:
```

**Review by TomWFox — APPROVED** [review]
Thanks for contributing, just a couple minor suggestions.

File: _includes/js/queries.md [hunk 3]
```diff
@@ -57,6 +57,27 @@ You can skip the first results by setting `skip`. In the old Parse hosted backen
 query.skip(10); // skip the first 10 results
 ```
 
+If you want to know the total number of rows in a table satisfying your query, for  e.g. pagination purposes - you can use `withCount` (`false` by default). 
+
+**Note:** Enabling this flag will change the structure of response, see the example below.
```
Comment (dplewis) [inline _includes/js/queries.md:62]: I thought it was true by default? You can just use query.withCount() no?

Comment (BufferUnderflower) [reply _includes/js/queries.md:62]: If you want to know the total number of rows in a table satisfying your query, for  e.g. pagination purposes - you can use `withCount()` (optionally accepts a boolean, defaults to `true` when called without arguments or `undefined`).
 
Will this be more clear?

Comment (dplewis) [reply _includes/js/queries.md:62]: (optionally accepts a boolean, defaults to true)  should be fine

Comment (BufferUnderflower) [reply _includes/js/queries.md:62]: can't this be mis-interpreted like it's an always enabled by default on every query?

Comment (dplewis) [reply _includes/js/queries.md:62]: @TomWFox What do you think?

Comment (TomWFox) [reply _includes/js/queries.md:62]: I agree with @dplewis, it's referencing that particular method not querying in general

Comment (TomWFox) [reply _includes/js/queries.md:62]: Alternatively, you could not mention the optional boolean as its not strictly relevant to this explanation.

Comment (BufferUnderflower) [reply _includes/js/queries.md:62]: right, jsdoc will show these details anyway when it comes to coding

Comment (TomWFox) [reply _includes/js/queries.md:62]: @dplewis are you happy with that change?

**Review by dplewis — APPROVED** [review]

**Comment by dplewis** [comment]
Thanks for contributing! I added this to the PHP SDK too.

**Comment by BufferUnderflower** [comment]
image
@dplewis Seems styling for markdown quotes is a bit too much

**Comment by dplewis** [comment]
Nice catch! I can take a look at it.

**Pull request merged**